UPGRADING.md

```diff
@@ -4,7 +4,264 @@ Upgrading Guide
+| `Bugsnag#beforeRecordBreadcrumb` | `Bugsnag#addOnBreadcrumb` |
+| `Bugsnag#clearBreadcrumbs` | `Bugsnag#addOnBreadcrumb/Configuration#setMaxBreadcrumbs` |
+| `Bugsnag#clearTab` | `Bugsnag#clearMetadata(String)` |
+| `Bugsnag#clearUser` | `Bugsnag#setUser` |
+| `Bugsnag#disableExceptionHandler` | `Configuration#setEnabledErrorTypes` |
+| `Bugsnag#enableExceptionHandler` | `Configuration#setEnabledErrorTypes` |
+| `Bugsnag#getMetaData`  | `Bugsnag#getMetadata` |
+| `Bugsnag#init(Context)` | `Bugsnag#start(Context)` |
+| `Bugsnag#init(Context, Configuration)` | `Bugsnag#start(Context, Configuration)` |
+| `Bugsnag#init(Context, String)` | `Bugsnag#start(Context, String)` |
+| `Bugsnag#init(Context, String, boolean)` | `Bugsnag#start(Context, Configuration)` |
+| `Bugsnag#leaveBreadcrumb(String, BreadcrumbType, Map<String,String>)` | `Bugsnag#leaveBreadcrumb(String, BreadcrumbType, Map<String,Object>)` |
+| `Bugsnag#notify(String, String, StackTraceElement[], Callback)` | `Bugsnag#notify(Throwable, OnErrorCallback)` |
+| `Bugsnag#notify(String, String, StackTraceElement[], Severity, MetaData)` | `Bugsnag#notify(Throwable, OnErrorCallback)` |
+| `Bugsnag#notify(String, String, String, StackTraceElement[], Severity, MetaData)` | `Bugsnag#notify(Throwable, OnErrorCallback)` |
+| `Bugsnag#notify(Throwable, Callback)` | `Bugsnag#notify(Throwable, OnErrorCallback)` |
+| `Bugsnag#notify(Throwable, MetaData)` | `Bugsnag#notify(Throwable, OnErrorCallback)` |
+| `Bugsnag#notify(Throwable, Severity)` | `Bugsnag#notify(Throwable, OnErrorCallback)` |
+| `Bugsnag#notify(Throwable, Severity, MetaData)` | `Bugsnag#notify(Throwable, OnErrorCallback)` |
+| `Bugsnag#stopSession` | `Bugsnag#pauseSession` |
+| `Bugsnag#setAppVersion` | `Configuration#setAppVersion` |
+| `Bugsnag#setAutoCaptureSessions` | `Configuration#setAutoTrackSessions` |
+| `Bugsnag#setBuildUUID` | `Configuration#setBuildUuid` |
+| `Bugsnag#setEndpoint` | `Configuration#setEndpoints` |
+| `Bugsnag#setErrorReportApiClient` | `Configuration#setDelivery` |
+| `Bugsnag#setFilters` | `Configuration#setRedactedKeys` |
+| `Bugsnag#setIgnoreClasses` | `Configuration#setIgnoreClasses` |
+| `Bugsnag#setLoggingEnabled` | `Configuration#setLogger` |
+| `Bugsnag#setMaxBreadcrumbs` | `Configuration#setMaxBreadcrumbs` |
+| `Bugsnag#setMetaData` | `Configuration#addMetadata` |
+| `Bugsnag#setNotifyReleaseStages` | `Configuration#setEnabledReleaseStages` |
+| `Bugsnag#setProjectPackages` | `Configuration#setProjectPackages` |
+| `Bugsnag#setReleaseStage` | `Configuration#setReleaseStage` |
+| `Bugsnag#setSendThreads` | `Configuration#setSendThreads` |
+| `Bugsnag#setSessionTrackingApiClient` | `Configuration#setDelivery` |
+| `Bugsnag#setUserEmail` | `Bugsnag#setUser` |
+| `Bugsnag#setUserId` | `Bugsnag#setUser` |
+| `Bugsnag#setUserName` | `Bugsnag#setUser` |
+
+See the [full documentation](https://docs.bugsnag.com/platforms/android) for more information.
+
+## Client
```

Comment:
I think we should keep this section because users of v4 will have been using `Client`.

I can update the suggested v5 API to point at the `Bugsnag` facade methods instead.